lib/ansible/playbook/task.py

```diff
@@ -106,6 +107,90 @@ def __init__(self, block=None, role=None, task_include=None):
 
         super(Task, self).__init__()
 
+    def _validate_module_defaults(self, attribute, name, value):
```

Comment:
This can probably be eliminated if the `resolve_group` logic is moved into `_load_module_defaults` on `Base`. It gets called *a lot* (for every task/host combo plus a few extras), and while it's short-circuiting a lot of the logic, I can't come up with a case where it's doing anything new over what would be done in `_load_module_defaults` (which is only called once per play/block/task that it's explicitly defined on).

Reply:
Good catch, thanks. I was testing this with one host and many tasks that had their own module_defaults and I didn't notice the difference.